src/main/java/org/prebid/server/bidder/smartrtb/SmartrtbBidder.java

```diff
@@ -119,73 +120,66 @@ private ExtImpSmartrtb parseImpExt(Imp imp) {
         }
     }
 
-    private static MultiMap resolveHeaders(Device device) {
-        final MultiMap headers = HttpUtil.headers()
-                .add("x-openrtb-version", "2.5");
-        if (device != null) {
-            HttpUtil.addHeaderIfValueIsNotEmpty(headers, "User-Agent", device.getUa());
-            HttpUtil.addHeaderIfValueIsNotEmpty(headers, "X-Forwarded-For", device.getIp());
-            HttpUtil.addHeaderIfValueIsNotEmpty(headers, "Accept-Language", device.getLanguage());
-
-            final Integer dnt = device.getDnt();
-            if (dnt != null) {
-                headers.add("DNT", dnt.toString());
-            }
-        }
-        return headers;
-    }
-
     @Override
     public Result<List<BidderBid>> makeBids(HttpCall<BidRequest> httpCall, BidRequest bidRequest) {
         if (httpCall.getResponse().getStatusCode() == HttpResponseStatus.NO_CONTENT.code()) {
             return Result.of(Collections.emptyList(), Collections.emptyList());
+        } else if (httpCall.getResponse().getStatusCode() == HttpResponseStatus.BAD_REQUEST.code()) {
+            return Result.emptyWithError(BidderError.badInput("Invalid request."));
+        } else if (httpCall.getResponse().getStatusCode() != HttpResponseStatus.OK.code()) {
+            return Result.emptyWithError(BidderError.badServerResponse(String.format("Unexpected HTTP status %s.",
+                    httpCall.getResponse().getStatusCode())));
         }
 
+        final BidResponse bidResponse;
         try {
-            final BidResponse bidResponse = Json.decodeValue(httpCall.getResponse().getBody(), BidResponse.class);
-            return extractBids(httpCall.getRequest().getPayload(), bidResponse);
-        } catch (DecodeException e) {
+            bidResponse = decodeBodyToBidResponse(httpCall);
+        } catch (PreBidException e) {
             return Result.emptyWithError(BidderError.badServerResponse(e.getMessage()));
         }
-    }
 
-    private static Result<List<BidderBid>> extractBids(BidRequest bidRequest, BidResponse bidResponse) {
-        if (bidResponse == null || bidResponse.getSeatbid() == null) {
-            return Result.of(Collections.emptyList(), Collections.emptyList());
+        final List<BidderBid> bidderBids = new ArrayList<>();
+        for (SeatBid seatBid : bidResponse.getSeatbid()) {
+            for (Bid bid : seatBid.getBid()) {
+                final ObjectNode ext = bid.getExt();
+                if (ext == null) {
+                    return Result.emptyWithError(BidderError.badServerResponse(String.format(
+                            "Invalid bid extension from endpoint.")));
+                }
+                try {
+                    final SmartrtbResponseExt smartrtbResponseExt = Json.mapper.treeToValue(ext,
+                            SmartrtbResponseExt.class);
```

Comment:
You can extract this and ext==null to another validation method, to make it cleaner